### PR TITLE
[BST-265] fix: missing found foreign record crashes records api call

### PR DESCRIPTION
**Pull request by adrianthedev**
# Description

When a foreign key record is missing the `/records` API call crashes.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Manual

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: features/data-sources/components/DataSourcesBlock.tsx [hunk 1]
```diff
@@ -65,7 +65,7 @@ const DataSourcesBlock = () => {
                     </div>
                     <br />
                     {organization?.name && (
-                      <div className="text-sm">{organization.name}</div>
+                      <div className="text-sm"><strong>Organization</strong>: {organization.name}</div>
```
Comment (adrianthedev) [inline features/data-sources/components/DataSourcesBlock.tsx:68]: This is just a tweak.

File: features/organizations/components/OrganizationsBlock.tsx [hunk 2]
```diff
@@ -69,7 +69,7 @@ const OrganizationsBlock = () => {
                   )}
                   {rolesByOrganizationId[org.id] && (
                     <div className="text-sm">
-                      Your role: {rolesByOrganizationId[org.id]}
+                      <strong>Your role</strong>: {rolesByOrganizationId[org.id]}
```
Comment (adrianthedev) [inline features/organizations/components/OrganizationsBlock.tsx:72]: This is just a tweak, unrelated to the fix

File: features/records/index.ts [hunk 3]
```diff
@@ -90,22 +91,25 @@ const hydrateAssociations = async (
       {
         columnName: "id",
         condition: "is_in",
-        value: uniq(foreignIds).toString(),
+        value: uniq(foreignIds).filter(Boolean).toString(),
```
Comment (adrianthedev) [inline features/records/index.ts:94]: Filter out `null` values. This was the first bug.

File: features/records/index.ts [hunk 4]
```diff
@@ -90,22 +91,25 @@ const hydrateAssociations = async (
...
+    );
 
     hydratedRecordsAssociation = hydratedRecordsAssociation.map(
       (record: any) => {
         const foreignRecordComputed = foreignRecords.find(
-          (foreignRecord: any) => foreignRecord.id === record[column.name]
+          (foreignRecord: any) => foreignRecord?.id === record[column.name]
         );
 
+        if (!foreignRecordComputed) return record;
```
Comment (adrianthedev) [inline features/records/index.ts:111]: Early return if the foreign record is not found. This was the second bug

File: plugins/fields/Association/helpers.ts [hunk 5]
```diff
@@ -19,11 +19,11 @@ export const getForeignName = (record: any, column: Column) => {
       if (record.title) prettyName = record.title;
       if (record.name) prettyName = record.name;
     }
-  }
 
-  if (prettyName) return `${prettyName} [${record.id}]`;
+    if (prettyName) return `${prettyName} [${record.id}]`;
```
Comment (adrianthedev) [inline plugins/fields/Association/helpers.ts:23]: These were outside of the `if(record)` conditional. They would have failed on `record.id`. This was the third bug.

**Review by adrianthedev — COMMENTED** [review]
Left some comments explaining the changes.

**Review by mihaimdm22 — APPROVED** [review]

**Pull request merged**